src/App.js

```diff
@@ -1,16 +1,46 @@
-import React from 'react';
-import './App.css';
+import React, { useState } from 'react';
+import ChatLog from './components/ChatLog';
 import chatMessages from './data/messages.json';
+import './App.css';
 
 const App = () => {
+  const initialCopy = chatMessages.map((chatEntry) => {
+    return { ...chatEntry };
+  });
+
+  const [chatData, setChatData] = useState(initialCopy);
+  const [likeCount, setLikeCount] = useState(0);
+
+  const updateLikes = (chatEntryId) => {
+    const updatedChatData = chatData.map((chatEntry) => {
+      if (chatEntry.id === chatEntryId) {
+        return {
+          ...chatEntry,
+          liked: !chatEntry.liked,
+        };
+      }
+      return chatEntry;
+    });
+    setChatData(updatedChatData);
+    updateLikeCount(updatedChatData);
```

Comment:
👀 If we use one of the above recommendations tocalculate the liked count directly from the chat data on render, we can omit this call, and the entire `updateLikeCount` function.